### ISSUE TITLE
jsx-indent Documentation Request

### ISSUE THREAD

**Issue by machineghost** [yannickcr/eslint-plugin-react#869]
I'm getting the following warning because of the `jsx-indent` rule:

```
Expected indentation of 10 space characters but found 12
```

This is confusing because I set the rule to 4 spaces:

```
"jsx-indent": ["warn", 4],
```

... and 10 is not a multiple of 4.

(**EDIT:** It turns out I had the rule wrong: it should be "react/jsx-indent".  However, I haven't closed the ticket because I still think an additional example would be helpful).

The problem is that I don't know whether the rule is being applied correctly or not because the current documentation only provides examples that start out with 0 indentation, and thus leaves ambiguity as to how code is handled when it starts out with indentation.

Also, none of the examples contain JS-in-JSX examples, so it's unclear whether those (eg. `{ foo }`) follow the JSX indentation rule or the ES Lint one (based on the errors I'm getting I think they follow the JSX one, but it would still be nice if the documentation could explain that).

A single extra example could clarify things:

```
// When is/isn't this valid?
class AppWrapper extends React.Component {
    render() {
        return <App>
            { 'Hello' }
        </App>
    }
}
```

Please consider adding the above example, or some similar one that starts out with indentation and contains a JS-in-JSX block, as I believe it would help disambiguate things and clarify how the rule works to anyone reading the docs.  Thanks.

**Comment by ljharb** [comment]
fwiw, you have `render` starting 5 spaces in, so that should be invalid with a setting of 4, but otherwise your suggested example is correct.

**Comment by lencioni** [comment]
It's almost correct. Render needs to return something.

**Comment by machineghost** [comment]
Heh, that's what I get for copying a real world example (which was originally an arrow function, thus the lack of a `return`).  I have edited my example to correct the indentation and add a `return`.

**Comment by ljharb** [comment]
It's now a valid example of correct indentation - altho with wrap-multilines, it'd be:
```jsx
// When is/isn't this valid?
class AppWrapper extends React.Component {
    render() {
        return (
            <App>
                {'Hello'}
            </App>
        );
    }
}
```
